primitives/blockchain/src/error.rs

```diff
@@ -163,4 +193,11 @@ impl Error {
 	pub fn from_state(e: Box<dyn sp_state_machine::Error>) -> Self {
 		Error::Execution(e)
 	}
+
+	/// Construct from a state db error.
+	// Can not be done directly, since that would make cargo run out of stack if
+	// `sc-state-db` is lib is added as dependency.
```

Comment:
```suggestion
```

Still weird. Is that some Cargo bug?

Reply:
I am not sure, I did not investigate that issue any further.